src/erlcloud_waf.erl

```diff
@@ -0,0 +1,1188 @@
+-spec update_size_constraint_set/3 ::
+       (ChangeToken :: string() | binary(),
+        SizeConstraintSetId :: string() | binary(),
+        Updates :: [waf_size_constraint_update()]) ->
+       waf_return_val().
+update_size_constraint_set(ChangeToken, SizeConstraintSetId, Updates) ->
+    update_size_constraint_set(ChangeToken, SizeConstraintSetId, Updates, default_config()).
+
+-spec update_size_constraint_set/4 ::
+       (ChangeToken :: string() | binary(),
+        SizeConstraintSetId :: string() | binary(),
+        Updates :: [waf_size_constraint_update()],
+        Config :: aws_config()) ->
+       waf_return_val().
+update_size_constraint_set(ChangeToken, SizeConstraintSetId, Updates, Config) ->
+    Json = [{<<"ChangeToken">>, to_binary(ChangeToken)},
+            {<<"SizeConstraintSetId">>, to_binary(SizeConstraintSetId)},
+            {<<"Updates">>, [transform_to_proplist(Update) || Update <- Updates]}],
+    waf_request(Config, "UpdateSizeConstraintSet", Json).
+
+%%%------------------------------------------------------------------------------
+%%% UpdateSqlInjectionMatchSet
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_UpdateSqlInjectionMatchSet.html
+%%%------------------------------------------------------------------------------
+-spec update_sql_injection_match_set/3 ::
+       (ChangeToken :: string() | binary(),
+        SqlInjectionMatchSetId :: string() | binary(),
+        Updates :: [waf_sql_injection_match_set_update()]) ->
+       waf_return_val().
+update_sql_injection_match_set(ChangeToken, SqlInjectionMatchSetId, Updates) ->
+    update_sql_injection_match_set(ChangeToken, SqlInjectionMatchSetId, Updates, default_config()).
+
+-spec update_sql_injection_match_set/4 ::
+       (ChangeToken :: string() | binary(),
+        SqlInjectionMatchSetId:: string() | binary(),
+        Updates :: [waf_sql_injection_match_set_update()],
+        Config :: aws_config()) ->
+       waf_return_val().
+update_sql_injection_match_set(ChangeToken, SqlInjectionMatchSetId, Updates, Config) ->
+    Json = [{<<"ChangeToken">>, to_binary(ChangeToken)},
+            {<<"SqlInjectionMatchSetId">>, to_binary(SqlInjectionMatchSetId)},
+            {<<"Updates">>, [transform_to_proplist(Update) || Update <- Updates]}],
+    waf_request(Config, "UpdateSqlInjectionMatchSet", Json).
+
+%%%------------------------------------------------------------------------------
+%%% UpdateWebACL 
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_UpdateWebACL.html
+%%%------------------------------------------------------------------------------
+-spec update_web_acl_default_action/3 ::
+       (ChangeToken :: string() | binary(),
+        WebAclId :: string() | binary(),
+        DefaultAction:: waf_acl_action_type()) ->
+       waf_return_val().
+update_web_acl_default_action(ChangeToken, WebAclId, DefaultAction) ->
+    update_web_acl_default_action(ChangeToken, WebAclId, DefaultAction, default_config()).
```

Comment:
`DefaultAction` can be renamed to `DefaultActionType` like [this](https://github.com/erlcloud/erlcloud/pull/259/files#diff-bb24148281aa8e81f08cbb145df91fc0R240)